### PR TITLE
handle npm aliased packages

### PR DESCRIPTION
**Pull request by cuixq**
Fixes https://github.com/google/osv-scanner/issues/588

### CODE REVIEW

**Review by another-rex — COMMENTED** [review]
LGTM, though can you add the test case in the issue where there's two dependencies of the same package (`string-width: ^5.1.2` and `string-width-cjs: npm:string-width@^4.2.0`)

File: pkg/lockfile/parse-npm-lock.go [hunk 1]
```diff
@@ -65,6 +68,13 @@ func parseNpmLockDependencies(dependencies map[string]NpmLockDependency) map[str
 		finalVersion := version
 		commit := ""
 
+		// If the package is aliased, get the name and version
+		if strings.HasPrefix(detail.Version, "npm:") {
+			i := strings.Index(detail.Version, "@")
```
Comment (michaelkedar) [inline pkg/lockfile/parse-npm-lock.go:73]: This will need to use `strings.LastIndex` to handle packages with scopes e.g. `npm:@foo/bar@1.2.3`

**Review by michaelkedar — COMMENTED** [review]
LGTM, just one thing

File: pkg/lockfile/parse-npm-lock.go (hunk 1 above)
Comment (cuixq) [reply pkg/lockfile/parse-npm-lock.go]: good point! I have also added a test case for this.

**Review by another-rex — APPROVED** [review]
LGTM, thanks!

**Review by michaelkedar — APPROVED** [review]

**Pull request merged**